lxd/storage/utils.go

```diff
@@ -218,6 +218,16 @@ func VolumeDBGet(pool *lxdBackend, projectName string, volumeName string, volume
 // VolumeDBCreate creates a volume in the database.
 // If volumeConfig is supplied, it is modified with any driver level default config options (if not set).
 func VolumeDBCreate(pool *lxdBackend, projectName string, volumeName string, volumeDescription string, volumeType drivers.VolumeType, snapshot bool, volumeConfig map[string]string, expiryDate time.Time, contentType drivers.ContentType) error {
+	// If the volumeType represents an instance type then check that the volumeConfig doesn't contain any of
+	// the instance disk effective override fields (which should not be stored in the database).
+	if volumeType.IsInstance() {
+		for _, k := range instanceDiskVolumeEffectiveFields {
+			if _, found := volumeConfig[k]; found {
```

Comment:
We usually split that:
```go
_, found := volumeConfig[k]
if found {
    // ...
}
```

Reply:
Yeah I've never really understood that for these bool cases where checking a map entry exists.
It makes sense when accessing the value of a key, e.g. `if foo := volumeConfig[k]; foo == "bar"` to a point (something about confused scoping and variable shadowing).
But for checking a field exists it seems like its more likely to introduce mistakes (incorrectly reusing a found var from another check) by creating a var outside of the scope of the if.

I've not gone strong opinions on it so will change though.

Reply:
I don't have strong opinions here either, but we should keep things consistent.

Reply:
@stgraber whats your thoughts on using `if _, found := volumeConfig[k]; found {` for checking if a map key exists (as opposed to extracting its value)?

Reply:
I've usually used the two lines syntax for that too. There's less of a concern about scoping that with errors but feels more consistent.

I tend to do:
```
_, ok := dict[k]
if ok {
}
```

`found` we've used as a generic boolean in other situations, `ok` we haven't and I believe matches the naming used in some of the upstream Go code so less likely to mask another variable.

Reply:
Hopefully we can use the generics in go 1.18 to create a KeyExists() function.